### PR TITLE
Docs and migration for 1.0.0

### PR DESCRIPTION
**Pull request by AsafMah**
### Added
- Migration guides
### Changed
- doc files are re-written to be more standard

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+Welcome to the migration guide for the Azure Data Explorer (Kusto) Go SDK. This guide is designed to assist you in migrating your application from using the beta version of the SDK (`github.com/Azure/azure-kusto-go`) to the new, stable release, version `1.0.0`.
+
+The release of version `1.0.0` introduces breaking changes, which include changes in dependencies, package arrangement, querying methods, and management commands. Following these steps carefully will ensure a smooth transition to the new version.
+
+## 1. Changes in Dependencies and go.mod File
+
+To migrate from the old SDK to the new one, you'll need to update your `go.mod` file's dependencies.
+
+**Old SDK:**
```
Comment (ohadbitt) [inline MIGRATION.md:11]: why needed ? can just put the new ..

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+```bash
+go get github.com/Azure/azure-kusto-go/azkustodata
+```
+
+- For ingestion operations (`github.com/Azure/azure-kusto-go/azkustoingest`):
+```bash
+go get github.com/Azure/azure-kusto-go/azkustoingest
+```
+
+Manually update your go.mod file by replacing `github.com/Azure/azure-kusto-go` with the specific package(s) you need.
```
Comment (ohadbitt) [inline MIGRATION.md:27]: Alternatively ?

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+**For Ingestion Operations:**
+```go
+import (
+    "github.com/Azure/azure-kusto-go/azkustoingest"
+)
+```
+
+### Using the New Packages
+
+Update your code to use the new packages.
```
Comment (ohadbitt) [inline MIGRATION.md:55]:
```suggestion
Update your code with the new package API
```

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+New SDK (For Data Operations):
+```go
+kustoConnectionStringBuilder := azkustodata.NewConnectionStringBuilder(endpoint)
+```
+
+Same for ingestion operations:
+
+Old SDK Ingestion Client Creation:
```
Comment (ohadbitt) [inline MIGRATION.md:71]:
```suggestion
Old SDK Ingestion Client Creation(Queued Example):
```

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+New SDK Ingestion Client Creation (Queued Example):
+```go
+in, err := azkustoingest.New(kustoConnectionString)
+```
+
+## 3. Building Queries
+
+The new SDK introduces a new way to build queries.:
```
Comment (ohadbitt) [inline MIGRATION.md:83]:
```suggestion
The new SDK introduces a new way to build queries:
```

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+	},
+)
+if err != nil {
+	panic("add error handling")
+}
+```
+
+The new SDK returns a dataset object, with the ability to handle different tables as they come:
+```go
+dataset, err := client.QueryIterative(ctx, "database", query)
```
Comment (ohadbitt) [inline MIGRATION.md:137]: So we want iterative to be the default user behavior ?

File: MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+if err != nil {
+    panic("error handling")
+}
+defer dataset.Close()
+
+for tableResult := range dataset.Tables() {
+    if tableResult.Err() != nil {
+        panic("table error handling")
+    }
+    for rowResult := range primaryResult.Table().Rows() {
```
Comment (ohadbitt) [inline MIGRATION.md:147]: primaryResult not set

File: MIGRATION.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+            panic("row error handling")
+        }
+        fmt.Println(rowResult.Row()) // Process each row
+    }
+}
+```
+
+You can also use the `Query` method to get all the results at once:
+```go
+dataset, err := client.Query(ctx, "database", query)
```
Comment (ohadbitt) [inline MIGRATION.md:158]: add here how to get the primary resultsl

File: MIGRATION.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+import github.com/Azure/azure-kusto-go/azkustoingest
+
+kcsb := azkustodata.NewConnectionStringBuilder(`endpoint`).WithAadAppKey("clientID", "clientSecret", "tenentID")
+ingestor, err := azkustoingest.New(kcsb, azkustoingest.WithDefaultDatabase("database"), azkustoingest.WithDefaultTable("table"))
+
+if err != nil {
+    // Handle error
+}
+
+defer ingestor.Close() // Always close the ingestor when done.
```
Comment (ohadbitt) [inline MIGRATION.md:216]: shouldnt we close if err != nil ?

File: README.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-# Microsoft Azure Data Explorer Public Preview (Kusto) [![GoDoc](https://godoc.org/github.com/Azure/azure-kusto-go?status.svg)](https://pkg.go.dev/github.com/Azure/azure-kusto-go/azkustodata) [![GoDoc](https://godoc.org/github.com/Azure/azure-kusto-go?status.svg)](https://pkg.go.dev/github.com/Azure/azure-kusto-go/azkustoingest)
+# Microsoft Azure Data Explorer (Kusto) [![GoDoc](https://godoc.org/github.com/Azure/azure-kusto-go?status.svg)](https://pkg.go.dev/github.com/Azure/azure-kusto-go/azkustodata) [![GoDoc](https://godoc.org/github.com/Azure/azure-kusto-go?status.svg)](https://pkg.go.dev/github.com/Azure/azure-kusto-go/azkustoingest)
```
Comment (ohadbitt) [inline README.md:1]: :)

File: MIGRATION.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
+# Migration Guide for Azure Data Explorer (Kusto) Go SDK
+
+Welcome to the migration guide for the Azure Data Explorer (Kusto) Go SDK. This guide is designed to assist you in migrating your application from using the beta version of the SDK (`github.com/Azure/azure-kusto-go`) to the new, stable release, version `1.0.0`.
+
+The release of version `1.0.0` introduces breaking changes, which include changes in dependencies, package arrangement, querying methods, and management commands. Following these steps carefully will ensure a smooth transition to the new version.
+
+## 1. Changes in Dependencies and go.mod File
+
+To migrate from the old SDK to the new one, you'll need to update your `go.mod` file's dependencies.
```
Comment (yogilad) [inline MIGRATION.md:9]:
```suggestion
To migrate to version 1.x from older versions, you'll need to update your `go.mod` file's dependencies.
```

File: MIGRATION.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+### Using the New Packages
+
+Update your code to use the new packages.
+
+For exmaple:
+
+Old SDK:
+```go
+kustoConnectionStringBuilder := kusto.NewConnectionStringBuilder(endpoint)
```
Comment (yogilad) [inline MIGRATION.md:61]: This is just a KCSB.
Do you want to show the client?

File: MIGRATION.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+
+## 3. Building Queries
+
+The new SDK introduces a new way to build queries.:
+The old SDK used a `kusto.NewStmt` method to build queries:
+```go
+    query := kusto.NewStmt("systemNodes | project CollectionTime, NodeId")
+```
+
+For the new SDK, use the `azkustodata/kql` package to build queries, which has a type-safe query builder:
```
Comment (yogilad) [inline MIGRATION.md:89]: is there an example of query parameters somewhere?

File: MIGRATION.md (hunk 6 above)
Comment (yogilad) [reply MIGRATION.md:137]: I'd show a non-iterative and an iterative example in that order

File: MIGRATION.md (hunk 12 above)
Comment (yogilad) [reply MIGRATION.md:61]: Also, is there value in showing these segmented line cases or do we want to focus on scenario examples (query, query with params, management command, ingest from , stream from, ... )

File: MIGRATION.md (hunk 3 above)
Comment (AsafMah) [reply MIGRATION.md:55]: I don't think this sentence is better, the point is the packages themselves are new

File: MIGRATION.md (hunk 9 above)
Comment (AsafMah) [reply MIGRATION.md:216]: If err != nil then the dataset is nil

File: MIGRATION.md (hunk 12 above)
Comment (AsafMah) [reply MIGRATION.md:61]: The idea is to demonstrate the new package name.

I wanted to give an overview of things they should change if they're migrating.
For more detailed examples they have the README and example files

File: MIGRATION.md (hunk 13 above)
Comment (AsafMah) [reply MIGRATION.md:89]: Yes in the README.md

Should it also be in the migration?

File: MIGRATION.md (hunk 6 above)
Comment (AsafMah) [reply MIGRATION.md:181]: Better?

File: MIGRATION.md (hunk 8 above)
Comment (AsafMah) [reply MIGRATION.md]: Rewrote

File: MIGRATION.md (hunk 1 above)
Comment (yogilad) [reply MIGRATION.md:11]: Very much, this being a migration guide.

File: MIGRATION.md (hunk 6 above)
Comment (yogilad) [reply MIGRATION.md:181]: Yes

File: MIGRATION.md (hunk 13 above)
Comment (yogilad) [reply MIGRATION.md:91]: only if there's a change to the class names or usage

**Pull request merged**